src/features/import-map.js

```diff
@@ -4,46 +4,90 @@
  * <script type="systemjs-importmap">{}</script>
  * OR
  * <script type="systemjs-importmap" src=package.json></script>
- * 
- * Only supports loading the first import map
  */
 import { baseUrl as pageBaseUrl, parseImportMap, resolveImportMap } from '../common.js';
 import { systemJSPrototype } from '../system-core.js';
 
-var importMap, importMapPromise;
+var importMapPromise, fetchPromises = {};
+
+// Start downloading the import maps as soon as possible
 if (typeof document !== 'undefined') {
-  const scripts = document.getElementsByTagName('script');
-  for (let i = 0; i < scripts.length; i++) {
-    const script = scripts[i];
-    if (script.type !== 'systemjs-importmap')
-      continue;
-
-    if (!script.src) {
-      importMap = parseImportMap(JSON.parse(script.innerHTML), pageBaseUrl);
-    }
-    else {
-      importMapPromise = fetch(script.src)
+  const scriptsToDownload = document.querySelectorAll('script[type="systemjs-importmap"][src]');
+
+  for (let i = 0; i < scriptsToDownload.length; i++) {
+    const script = scriptsToDownload[i]
+    fetchPromises[script.src] = fetch(script.src)
```

Comment:
Rather than writing fetch promises into this object, can we just do:

```js
importMapPromise = importMapPromise.then(function () {
  return fetch(script.src).then(...);
});
```

Reply:
This code path could also then become:

```js
if (script.src) /// script src path
else // inline path
```

Reply:
See my previous comment with the question about when the document order of import maps should take effect.

Reply:
I may be reading this wrong, but would that mean that you can't fetch the next import map until the previous one is done downloading? (assuming you have two or more external import maps.)

Whereas the current way downloads them concurrently and then merges them in order if I'm reading it correctly.

But I guess your suggestion does keep it more in line with the spec, right?

Reply:
Ah that's a good point -- it would delay parallel downloading of import maps.